src/makeInfinityAdder.js

```diff
@@ -6,7 +6,30 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let itsFirstCall = true;
+
+  const adder = num => {
+    if (num === undefined && itsFirstCall) {
+      itsFirstCall = false;
+      sum = 0;
+
+      return sum;
+    }
+
+    if (num === undefined) {
+      itsFirstCall = true;
+
+      return sum;
+    }
```

Comment:
generally, you don't have to check for the first call, just preserve current total, return it, and don't forget to annulate `sum`